sql/wsrep_mysqld.cc

```diff
@@ -2294,6 +2296,17 @@ int wsrep_to_isolation_begin(THD *thd, const char *db_, const char *table_,
                              const TABLE_LIST* table_list,
                              Alter_info* alter_info, wsrep::key_array* fk_tables)
 {
+  /*
+    MDEV-32738: Check if DDL operation has been killed before.
+  */
+  mysql_mutex_lock(&thd->LOCK_thd_kill);
+  const killed_state killed = thd->killed;
+  mysql_mutex_unlock(&thd->LOCK_thd_kill);
+  if (killed) {
```

Comment:
This has still race-condition. Let's assume this DDL-operation is not yet killed but killed before we start it on wsrep-layer. We might need to keep mutex above longer (not sure if it's safe...).

Reply:
I think after `close_thread_tables()` call has been performed in `wsrep_append_fk_parent_table()` it's safe to assume that no-one can kill this operation by BF-abort afterwards (MDL locks should have been released by that point).

I will add more comments to explain that.

Reply:
To make this clearer, you could perhaps add the check for thd->killed in wsrep_append_fk_parent_table()?
We can leave the check in wsrep_to_isolation_begin(), as a "catch all" kind of condition.